controller/backing_image_controller.go

```diff
@@ -401,6 +401,20 @@ func (bic *BackingImageController) handleBackingImageDataSource(bi *longhorn.Bac
 			if err != nil {
 				return err
 			}
+
+			// For cloning, we choose the same node and disk as the source backing image
+			if bi.Spec.SourceType == longhorn.BackingImageDataSourceTypeClone {
+				sourceBackingImageName := bi.Spec.SourceParameters[longhorn.DataSourceTypeCloneParameterBackingImage]
+				sourceBackingImage, err := bic.ds.GetBackingImageRO(sourceBackingImageName)
+				if err != nil {
+					return fmt.Errorf("failed to get source backing image %v during cloning", sourceBackingImageName)
+				}
+				readyNode, readyDiskName, err = bic.ds.GetOneBackingImageReadyNodeDisk(sourceBackingImage)
+				if err != nil {
+					return fmt.Errorf("failed to find one ready source backing image %v during cloning", sourceBackingImageName)
+				}
```

Comment:
Move the codes to a helper function, e.g.

```
func findReadyNodeAndDisk(...) (nodeName, diskName string, err error) {
    ...
}
```

Reply:
fixed.